src/awx_plugins/credentials/azure_mi.py

```diff
@@ -0,0 +1,60 @@
+# FIXME: the following violations must be addressed gradually and unignored
+# mypy: disable-error-code="import-untyped, no-untyped-def"
```

Comment:
Try to avoid having ignored checks in freshly added modules. This was a band-aid for pre-existing code, but for new code, it shouldn't be allowed to have such global ignores. If there's places that generate errors, add `# type: ignore[error-code]` inline and a comment explaining what's wrong and when it can be dropped. Global ignores allow more errors to sneak in over time.

Reply:
I've merged the new work into an existing plugin and tried to address the linting issues in that plugin.